hw/vendor/lint/pulp_riscv_dbg.vlt

```diff
@@ -62,3 +62,14 @@ lint_off -rule WIDTH -file "*/src/dm_mem.sv" -match "*SUB expects 32 bits*'DataC
 // this might actually be Verilator doing something a bit strange.
 lint_off -rule WIDTH -file "*/src/dm_mem.sv" -match "*SUB expects 32*but*SEL*10 bits."
 
+// Avoid a warning about address truncation. These addresses are both derived
+// from the 64-bit dm::HaltAddress. They get assigned to something with
+// DbgAddressBits (12) bits. Both HaltAddress and DbgAddressBits come from the
+// original code (neither is a parameter that is being passed in from
+// OpenTitan)
+lint_off -rule WIDTH -file "*/src/dm_mem.sv" -match "Operator VAR 'Rom*Addr' expects 12 bits*generates 64 bits."
```

Comment:
We discussed this [here](https://github.com/lowRISC/opentitan/pull/21736#discussion_r1507261569), right?

> This only holds as long as `dm::HaltAddress` in `dm_pkg.sv` doesn't get changed. It feels wrong to add this waiver, which is decoupled from `dm_pkg.sv`. Wouldn't it be better to fix the RTL, so that only the 12 LSB of `dm::HaltAddress` are used to compute `RomBaseAddr` and `RomEndAddr`? We could do that with a patch while we create an upstream PR and wait for it to get approved (or not, in which case we keep the patch).

Why waive this instead of patching the RTL?

Reply:
The honest answer: because the RTL is vendored! Would it make sense to waive this locally but also raise an issue in the pulp project (and we can then include the pulp issue ID in the waiver line)?

Reply:
I think this lint warning is not critical, so I don't put a veto on adding this waiver. I'd prefer getting the RTL fixed, though, but probably we have more important things to do right now.

So please feel free to go ahead with adding this wavier, but then I suggest creating a D3 issue to fix this in RTL and remove this waiver.

Reply:
Issue created: #22545.